core/src/state/splitted_state.rs

```diff
@@ -24,6 +25,21 @@ pub struct SplittedWriter<O, R, W> {
   batched_modify: Sc<Cell<ModifyScope>>,
   create_at: Instant,
   last_modified: Sc<Cell<Instant>>,
+  ref_count: Sc<Cell<usize>>,
+}
+
+impl<O, R, W> Drop for SplittedWriter<O, R, W> {
+  fn drop(&mut self) {
+    if self.ref_count.get() == 1 {
+      let mut notifier = self.notifier.clone();
+      // we use an async task to unsubscribe to wait the batched modifies to be
+      // notified.
+      AppCtx::spawn_local(async move {
```

Comment:
unnecessary batch unsubscribe, it should only be called once.

Reply:
![image](https://github.com/RibirX/Ribir/assets/6829874/0957c6ea-f3d5-414c-acb4-d063a0c90b9b)
but it's async unsubscribe at drop of stateful. why should they be different? or just both should be called not batch

Reply:
My bad, the comments already explained everything